src/gr-convert-units.c

```diff
@@ -422,11 +422,129 @@ gr_convert_human_readable (double *amount, GrUnit *unit)
                 ;
     }
 
-                if (*unit == unit1) {
-                unit_changed = FALSE; 
+    if (*unit == unit1) {
+    unit_changed = FALSE; 
+    }
+    
+    *amount = amount1;
+    *unit = unit1;
+    }
+}
+
+void
+gr_convert_multiple_units (double *amount1, GrUnit *unit1, double *amount2, GrUnit *unit2)
+{
+
+    double n1 = *amount1;
+    double n2 = *amount2;
+    GrUnit u1 = *unit1;
+    GrUnit u2 = *unit2;
+    double fractional, integer;
+
+    gr_convert_human_readable(&n1, &u1);
+
+    fractional = modf(n1, &integer);
+
+    if (u1 != GR_UNIT_UNKNOWN) {
+            if (fractional > 0)
+            {
+                n2 = fractional;
+                u2 = u1;
+                gr_convert_human_readable(&n2, &u2);
+
+                if (u1 != u2)
+                {
+                        n1 = integer;
                 }
                 
-                *amount = amount1;
-                *unit = unit1;
+                else 
+                {
+                        n2 = 0;
+                        u2 = GR_UNIT_UNKNOWN;
+                }
+        }
+        
+        else
+        {
+                n2 = 0;
+                u2 = GR_UNIT_UNKNOWN;
+        }
+    }
+
+    else
+    {
+            n2 = 0;
+            u2 = GR_UNIT_UNKNOWN;
+    }
+
+    *amount1 = n1;
+    *unit1 = u1;
+    *amount2 = n2;
+    *unit2 = u2;
+}
+
+void
+gr_convert_format_for_display  (GString *s, double a1, GrUnit u1, double a2, GrUnit u2) 
+{
+
+    if (u1 == GR_UNIT_UNKNOWN || (!u1)) {
```

Comment:
the !ui check doesn't really make sense.
Effectively, that is a check for u1 == 0, which is the same as u1 == GR_UNIT_UNKNOWN